shared-libs/cht-datasource/src/contact.ts

```diff
@@ -0,0 +1,152 @@
+   * @param context the current data context
+   * @returns a function for retrieving a contact with the contact's parent lineage
+   * @throws Error if a data context is not provided
+   */
+  /**
+   * Returns a contact for the given qualifier with the contact's parent lineage.
+   * @param qualifier identifier for the contact to retrieve
+   * @returns the contact or `null` if no contact is found for the qualifier
+   * @throws Error if the qualifier is invalid
+   */
+  export const getWithLineage = getContact(Local.Contact.v1.getWithLineage, Remote.Contact.v1.getWithLineage);
+
+  /**
+   * Returns a function for retrieving a paged array of contact identifiers from the given data context.
+   * @param context the current data context
+   * @returns a function for retrieving a paged array of contact identifiers
+   * @throws Error if a data context is not provided
+   * @see {@link getIdsAll} which provides the same data, but without having to manually account for paging
+   */
+  export const getIdsPage = (context: DataContext): typeof curriedFn => {
+    assertDataContext(context);
+    const fn = adapt(context, Local.Contact.v1.getPage, Remote.Contact.v1.getPage);
+
+    /**
+     * Returns an array of contact identifiers for the provided page specifications.
+     * @param qualifier the limiter defining which identifiers to return
+     * @param cursor the token identifying which page to retrieve. A `null` value indicates the first page should be
+     * returned. Subsequent pages can be retrieved by providing the cursor returned with the previous page.
+     * @param limit the maximum number of identifiers to return. Default is 10000.
+     * @returns a page of contact identifiers for the provided specification
+     * @throws InvalidArrgumentError if no qualifier is provided or if the qualifier is invalid
+     * @throws InvalidArgumentError if the provided `limit` value is `<=0`
+     * @throws InvalidArgumentError if the provided cursor is not a valid page token or `null`
+     */
+    const curriedFn = async (
+      qualifier: ContactTypeQualifier | FreetextQualifier,
+      cursor: Nullable<string> = null,
+      limit = 100
+    ): Promise<Page<string>> => {
+      assertCursor(cursor);
+      assertLimit(limit);
+
+      if (ContactTypes.v1.isContactType(qualifier)) {
+        assertTypeQualifier(qualifier);
+      }
+
+      if (ContactTypes.v1.isFreetextType(qualifier)) {
+        assertFreetextQualifier(qualifier);
+      }
+
+      return fn(qualifier, cursor, limit);
+    };
+    return curriedFn;
+  };
+
+  /**
+   * Returns a function for getting a generator that fetches contact identifiers from the given data context.
+   * @param context the current data context
+   * @returns a function for getting a generator that fetches contact identifiers
+   * @throws Error if a data context is not provided
+   */
+  export const getIdsAll = (context: DataContext): typeof curriedGen => {
```

Comment:
@sugat009 Looking at this again, I think we used the `getAll` naming for person/place because the alternative was just something redundant like `Person.getPersons`.  For the `ids` functions, I think it makes sense to go with `getIds`.  How does that sound do you?

Reply:
@jkuester sounds good to me.